tests/datavolume_test.go

```diff
@@ -857,30 +857,61 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 		})
 	})
 
-	Describe("Create/Delete same datavolume in a loop", func() {
+	Describe("Create/Delete/Create/Delete same datavolume in a loop", func() {
 		Context("retry loop", func() {
-			dataVolumeName := "dv1"
 			numTries := 5
-			for i := 1; i <= numTries; i++ {
-				It(fmt.Sprintf("[test_id:3939][test_id:3940][test_id:3941][test_id:3942][test_id:3943]should succeed on loop %d", i), func() {
+			It(fmt.Sprintf("[test_id:3939][test_id:3940][test_id:3941][test_id:3942][test_id:3943] should succeed on %d loops", numTries), func() {
+				dataVolumeName := "test-dv"
+				dataVolumeNamespace := f.Namespace
+				for i := 1; i <= numTries; i++ {
+					By(fmt.Sprintf("running loop %d", i))
 					url := fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs)
-					dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, "1Gi", url)
+					dv := utils.NewDataVolumeWithHTTPImport(dataVolumeName, "1Gi", url)
 
-					By(fmt.Sprintf("creating new datavolume %s", dataVolume.Name))
-					dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+					By(fmt.Sprintf("creating new datavolume %s", dataVolumeName))
+					dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, dataVolumeNamespace.Name, dv)
 					Expect(err).ToNot(HaveOccurred())
 					f.ForceBindPvcIfDvIsWaitForFirstConsumer(dataVolume)
 
+					By("verifying pvc was created and is bound")
+					pvc, err := utils.WaitForPVC(f.K8sClient, dataVolume.Namespace, dataVolume.Name)
+					Expect(err).ToNot(HaveOccurred())
+					//We use the PVC UID to confirm later a new PVC was created
+					pvcUID := pvc.UID
+
 					By(fmt.Sprintf("waiting for datavolume to match phase %s", cdiv1.Succeeded))
-					err = utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, dataVolume.Name)
+					err = utils.WaitForDataVolumePhase(f.CdiClient, dataVolume.Namespace, cdiv1.Succeeded, dataVolume.Name)
 					Expect(err).ToNot(HaveOccurred())
 
 					By("deleting DataVolume")
-					err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, dataVolumeName)
+					err = utils.DeleteDataVolume(f.CdiClient, dataVolume.Namespace, dataVolume.Name)
 					Expect(err).ToNot(HaveOccurred())
 
-				})
-			}
+					//Create the DV again to test if a PVC collision with a terminating PVC does not reject the DV creation
+					By("verifying if the DV creation is not rejected if a terminating PVC with same name exists")
```

Comment:
Maybe I am missing something, but the whole point of the loop is so we don't have to do anything that is below the delete of the datavolume.
So the following happens in a loop:
1. Create datavolume (it could be it still exists from previous loop)
2. Wait for datavolume to be complete (this is important, it stops us from running through the entire loop immediately)
3. Delete datavolume, but do NOT verify it is gone.
4. Goto 1.
The first iteration of the loop should be like before, but the second iteration 1. should NOT get rejected, but 2 will not happen until the original PVC is actually gone, and a new one is created. So doing it several times in a row will verify the feature is working as expected.

Reply:
You are right, we do not need to have the whole verification within one loop, but only verify the collision starting from the second iteration. I updated that.